### DISCUSSION TITLE
데이터 타입의 종류는 어떤 것들이 있나요?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#4] [데이터 타입]
- 데이터 타입에는 무엇이 있나요?
  - 원시 타입과 객체(참조) 타입의 차이점에 대해 설명해주세요.
  - 자바스크립트의 Number 타입은 다른 언어들과 차이점이 무엇인가요?
  - 심볼 타입에 대해 설명해주세요.
  - JavaScript에서 템플릿 리터럴이란 무엇인가요?
    - 문자열 타입인 템플릿 리터럴의 장점을 설명해주세요.
      - 표현식을 삽입할 때 템플릿 리터럴을 사용하지 않고 문자열을 삽입하는 방법에 대해 말해주세요.
  - undefined와 null의 차이는 무엇인가요?

**Comment by publdaze** [comment]
> - 데이터 타입에는 무엇이 있나요?
  - 원시타입, 객체 타입으로 구분
  - 원시타입은 Number, String, Boolean, null, undefined, Symbol, BigInt이 있음
  - 객체 타입은 함수, 배열, 객체 등 원시타입이 아닌 타입
> - 숫자 타입은 정수, 실수등으로 나뉘나요?
  - 64비트 부동소수점 형식을 따름 -> 정수, 실수 구분 없이 모두 실수로 처리
> - 심볼 타입에 대해 설명해주세요.
  - 변경 불가한 원시 값
  - 중복 불가
  - 객체 속성 중복을 허용하지 않기 위해 주로 사용
> - 문자열 타입인 템플릿 리터럴의 장점을 설명해주세요.
  - 멀티 라인
  - 표현식 삽입
  - 태그드 템플릿
> - undefined와 null의 차이는 무엇인가요?
  - 서로 다른 데이터 타입
  - undefined는
    - 변수 선언 시 자바스크립트 엔진에 의해 초기화할 때 사용되는 값
  - null은
    - 개발자가 변수에 값이 없음을 의도적으로 명시할 때 사용

**Comment by pipisebastian** [comment]
> 데이터 타입에는 무엇이 있나요? 원시 타입과 객체(참조) 타입의 차이점에 대해 설명해주세요.

1. 원시 타입(`변경불가능`) : 숫자(number), 문자열(string), 불리언(boolean), undefined, null, symbol, BigInt
2. 참조 타입(`변경 가능`) : 원시타입을 제외한 객체, 배열, 함수 ...

> 자바스크립트의 Number 타입은 다른 언어들과 차이점이 무엇인가요?

정수, 실수를 구분하지않고 모두 실수로 처리가 됩니다.

> 심볼 타입에 대해 설명해주세요.

객체 프로퍼티를 고유값으로 사용할 수 있습니다.

>javaScript에서 템플릿 리터럴이란 무엇인가요?  문자열 타입인 템플릿 리터럴의 장점을 설명해주세요.

백틱을 사용하여 문자열을 표시하는데, 백틱내부에서 줄바꿈이 적용되며, 표현식을 삽입할 수 있기에, 일반 따옴표, 쌍따옴표보다 가독성이 증가합니다.

> 표현식을 삽입할 때 템플릿 리터럴을 사용하지 않고 문자열을 삽입하는 방법에 대해 말해주세요.

1. `+` 연산자
2. **`String`** 객체의 **`concat`** 메서드
4. 배열의 **`join`** 메서드

> undefined와 null의 차이는 무엇인가요?

`undefined`는 자바스크립트 엔진이 변수를 초기화할때 사용합니다. 그렇기에 의도적으로 개발자가 값이 없음을 나타내려면 `null` 을 사용하면 됩니다.
`null` 이 담긴 값은 은 추후 메모리에서 해제됩니다.
